src/RA_Defs.cpp

```diff
@@ -91,19 +173,77 @@ void RADebugLog(const char* format, ...)
     *p++ = '\n';
     *p = '\0';
 
-    OutputDebugString(NativeStr(buf).c_str());
-
-    //SetCurrentDirectory( g_sHomeDir.c_str() );//?
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, RA_LOG_FILENAME, "a") == 0)
-    {
-        fwrite(buf, sizeof(char), strlen(buf), pf);
-        fclose(pf);
-    }
+    RADebugLogNoFormat(buf);
 }
 
+_Use_decl_annotations_
 BOOL DirectoryExists(const char* sPath)
 {
     DWORD dwAttrib = GetFileAttributes(NativeStr(sPath).c_str());
     return(dwAttrib != INVALID_FILE_ATTRIBUTES && (dwAttrib & FILE_ATTRIBUTE_DIRECTORY));
 }
+
+_Use_decl_annotations_
+ResizeContent::ResizeContent(HWND parentHwnd, HWND contentHwnd, AlignType newAlignType, bool isResize) noexcept
+{
+    hwnd.reset(contentHwnd);
+    nAlignType = newAlignType;
+    bResize = isResize;
+
+
+    auto lpRect{ std::make_unique<RECT>() };
+    GetWindowRect(hwnd.get(), lpRect.get());
```

Comment:
I was experimenting between `std::make_unique<RECT>()` and `std::make_unique<RARect>()`. As far as I can tell, they have the same effect. I made compile-time be removing inheritance and using a conversion operator.